### DISCUSSION TITLE
Sunbird-Knowlg: Release-5.1.0 kick-start

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Knowlg/Community#42] [General Discussions]
We are planning to kickstart release-5.1.0. here are the details
  | Start date | End date
-- | -- | --
Planning phase | 18 Jul 2022 | 28 Jul 2022
Sprint 1 | 01 Aug 2022 | 19 Aug 2022
Sprint 2 | 22 Aug 2022 | 09 Sep 2022
PPV | 12 Sep 2022 | 23 Sep 2022
Prod | 26 Sep 2022

Ease of Adoption | Priority |  
-- | -- | --
API automantion - using Postman - Content-service -> Scanesios & Enahcement of +vs & -ve tests - Search service -> API documentation & tests - DIAL service -> Scanesios & Enahcement of +vs & -ve tests | P1 | https://project-sunbird.atlassian.net/browse/KN-15 https://project-sunbird.atlassian.net/browse/KN-14
Schema externalization | P2 | https://project-sunbird.atlassian.net/browse/KN-16
Neo4J v3.3 alternative | P1 | https://project-sunbird.atlassian.net/browse/KN-1
Sunbird-knowlg Portal - V2 | P2 |  
Java version upgrade(from 8 to 11)- Content Publish API - Taxonomy APIs | P1 | https://project-sunbird.atlassian.net/browse/KN-9https://project-sunbird.atlassian.net/browse/KN-10
Angular Upgrade to 10 - Collection-Editor - Players-V2 | P1 | https://project-sunbird.atlassian.net/browse/KN-20https://project-sunbird.atlassian.net/browse/KN-26

BB truly independent | Priority |  
-- | -- | --
Design: Separation of of question-set & Collection-Editor - Part 2 | P2 | https://project-sunbird.atlassian.net/browse/SB-28699
Mobile App: To test players | P1 |  
Sunbird-video-player: Publishing as NG module | P2 | https://project-sunbird.atlassian.net/browse/SB-29186
Workspace moving from Sunbird to VDN - From CoCreate | P2 |  

Cloud Agnostic(CSP) |  Prioirty |  
-- | -- | --
Knowlg microservices, editors & players should be cloud agnostic | P1 |  

Please share your comments/feedback on the priority of the items. If any other priority items are missing in the above list please share the details in the below comments. We will try to accommodate if time permits.

I would request the contributors also share the list of items if you want to push any features within the above release timelines.

@pramodkvarma, @vrayulu, @SanthoshVasabhaktula, @rhwarrier,
@rahul-kankaria, @ChakshuGautam, @kiranharidas187, @pritamps, @chitranshu-keshav @rahul101001000 @coolbung
@surendrasinghs, @pallakartheekreddy, @krgauraw, @luckynvdu @reshmi-nair, @sharathkashyap, @gandham-santhosh, @kameshbhr, @AmiableAnil, @Jayaprakash8887, @anandp504,
@surendrasinghs

Shikshalokam: @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL
Samagra: @Prerna789 @aayushkedia95
SunbirdCB: @pkranga
Avanti: @pritamps @karn-nahata
DC's: @SanthoshVasabhaktula @vrayulu @maheshkumargangula
Sunbird QA: @ashokreddy1208
@gandham-santhosh
@kameshbhr @mmanimegalai
@madhucr @alok-os

**Comment by vinukumar-vs** [comment]
Due to a lot of unknowns to making Knolwg BB products(microservice, Editors) cloud-agnostic (for CSP migration) took more time than the estimated time. Hence below items got deprioritized from release-5.1.0

Ease of Adoption | Priority |  
-- | -- | --
API automantion - using Postman - Content-service -> Scanesios & Enahcement of +vs & -ve tests - Search service -> API documentation & tests - DIAL service -> Scanesios & Enahcement of +vs & -ve tests | P1 | https://project-sunbird.atlassian.net/browse/KN-15 https://project-sunbird.atlassian.net/browse/KN-14
Schema externalization | P2 | https://project-sunbird.atlassian.net/browse/KN-16
Neo4J v3.3 alternative | P1 | https://project-sunbird.atlassian.net/browse/KN-1

BB truly independent | Priority |  
-- | -- | --
Design: Separation of of question-set & Collection-Editor - Part 2 | P2 | https://project-sunbird.atlassian.net/browse/SB-28699
Mobile App: To test players | P1 |  
Workspace moving from Sunbird to VDN - From CoCreate | P2 |  

@rhwarrier, @chitranshu-keshav

**Reply by rhwarrier** [reply]
@vinukumar-vs  we'll consider these as the first bucket for 5.2

**Reply by vinukumar-vs** [reply]
@rhwarrier First sprint of Release-5.2 is still occupied with CSP migration tasks. We can take these in sprint-2 of 5.2 only if CSP migration activities are on track as per plan.

**Comment by Priyakashi** [comment]
@vinukumar-vs As discussed, there would be a delay in regression execution and eventually a delay of 4 days to give Knowlg BB sign off. Sign off would be provided for 5.1.0 on 30-Sept.
@rhwarrier @ShailandraOjha @chitranshu-keshav @anunagaraj @luckynvdu @sharathkashyap

**Comment by Priyakashi** [comment]
@vinukumar-vs Due to an interrupt of SB-30516 test case execution, there would be a delay of 2 days in giving sign off for 5.1.0. Sign off would be provided for 5.1.0 on 6-Oct (4th and 5th holiday).
@rhwarrier @ShailandraOjha @chitranshu-keshav @anunagaraj @luckynvdu @sharathkashyap

**Reply by anunagaraj** [reply]
@vinukumar-vs [Priyakashi](https://github.com/Priyakashi) Did we sign off R 5.1.0.KIndly share an update .
@rhwarrier @ShailandraOjha @chitranshu-keshav@luckynvdu @sharathkashyap

**Comment by anunagaraj** [comment]
@vinukumar-vs @Priyakashi Please share update on R5.1.0 Sign off.
@rhwarrier @ShailandraOjha @chitranshu-keshav@luckynvdu @sharathkashyap

**Comment by Priyakashi** [comment]
@anunagaraj R5.1.0 Sign off is provided, microsite will be updated accordingly.
cc: @vinukumar-vs 
@rhwarrier @ShailandraOjha @chitranshu-keshav@luckynvdu @sharathkashyap
